### ISSUE TITLE
Android Hybrid - Search - App crashes when switching between expenses of the same report.

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#51584]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.54-2
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/5135640&group_by=cases:section_id&group_order=asc&group_id=296775
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Open the Expensify app
2. Open the search section
3. Tap on "Outstanding"
4. Open any report with several expenses on it. (At least 3 or 4)
5. Open any expense on it
6. Tap on the arrow on the top left corner
7. Open another expense
8. Repeat steps 5 to 7 a few more times
9. Verify that you are able to switch between expenses without the app closing

## Expected Result:
The user should be able to switch between expenses of the same report, all the necessary times, without the app closing

## Actual Result:
The app crashes when switching between expenses of the same report in "Outstanding" section in search

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [x] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/3dc1b552-2f44-469c-b17b-72be0f9e2aa8

[logs (2).txt](https://github.com/user-attachments/files/17546130/logs.2.txt)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
Production:

https://github.com/user-attachments/assets/aec17a6b-5528-4a58-b7f5-ee73e9d8551d

**Comment by lanitochka17** [comment]
We think that this bug might be related to #wave-collect - Release 1

**Comment by jasperhuangg** [comment]
At the dentist, will tackle this when I'm back in a few hours

**Comment by jasperhuangg** [comment]
Building a revert that might be related

**Comment by hannojg** [comment]
@lanitochka17 for debugging this issue, are you able to reproduce the bug on your device, connect it to a computer and run `adb bugreport` from a terminal and share the zip file here with us? 

(The crash from the logs is an ANR, and in that log file we have no info to what happened. This bugreport zip file will help us to understand what caused the crash)

**Comment by jasperhuangg** [comment]
@hannojg I'm testing right now, I'll try that out

**Comment by hannojg** [comment]
Thanks! (Btw, the pr hasn't touched any UI related stuff and nothing in regards to reports, so i am not sure how likely it is related - but lets see!)

**Comment by jasperhuangg** [comment]
[dumpstate-2024-10-28-11-25-07.zip](https://github.com/user-attachments/files/17547292/dumpstate-2024-10-28-11-25-07.zip)

I was able to reproduce on staging, here's the `dumpstate`, although I'm not entirely sure if the command succeeded on my device @hannojg

**Comment by jasperhuangg** [comment]
I was also able to reproduce the bug in the NewDot App (not HybridApp), so we should be able to test some reverts.

**Comment by jasperhuangg** [comment]
Yeah seems like the `adb bugreport` command is having permissions errors, looking into it.

**Comment by jasperhuangg** [comment]
[bugreport-beyond1ltexx-SP1A.210812.016-2024-10-28-11-43-45.zip](https://github.com/user-attachments/files/17547476/bugreport-beyond1ltexx-SP1A.210812.016-2024-10-28-11-43-45.zip)

Here's the bug report after running it on New Expensify

**Comment by jasperhuangg** [comment]
```
10-28 11:24:45.415  1058   500   500 I tombstoned: received crash request for pid 3837
10-28 11:24:45.415  1058   500   500 I tombstoned: found intercept fd 512 for pid 3837 and type kDebuggerdJavaBacktrace
10-28 11:24:45.416 10326  3837  3842 I putmethod.lati: Wrote stack traces to tombstoned
10-28 11:24:45.416  1058   500   500 W tombstoned: missing output fd
10-28 11:24:45.418  1000   971 32163 I system_server: libdebuggerd_client: done dumping process 3837
10-28 11:24:45.425  1000   971 32163 I ActivityManager: Collecting stacks for pid 3968
10-28 11:24:45.430  1000   971 32163 I system_server: libdebuggerd_client: started dumping process 3968
10-28 11:24:45.431  1058   500   500 I tombstoned: registered intercept for pid 3968 and type kDebuggerdJavaBacktrace
10-28 11:24:45.433  1000  3968  3973 I c.android.sdhm: Thread[6,tid=3973,WaitingInMainSignalCatcherLoop,Thread*=0x7b93c50f50,peer=0x13240d60,"Signal Catcher"]: reacting to signal 3
10-28 11:24:45.433  1000  3968  3973 I c.android.sdhm: 
```

Doesn't seem to be very useful 🤔

**Comment by jasperhuangg** [comment]
Waiting on app deploys to be fixed before CPing

**Comment by jasperhuangg** [comment]
Retested and seems to have fixed the issue, navigated to/from expenses in the report a bunch of times and no crash.

**Comment by hannojg** [comment]
> I reproduced the bug on NewDot staging after navigating to/back from roughly 5-10 expense reports.

@jasperhuangg were all those expense reports part of the same chat/report?

**Comment by hannojg** [comment]
This is so weird. The code I have added is only used in `SearchRouter`. The code i added isn't even invoked in the flow that this bug describes.

**Comment by hannojg** [comment]
Looking at the ANRs from your device we can see that the main/UI and JS thread are in a deadlock, presumably due to some sync calls caused by reanimated:

| JS Thread | Main Thread |
|--------|--------|
| ![CleanShot 2024-10-29 at 15 26 44](https://github.com/user-attachments/assets/9a2e0cab-c6d0-4304-987a-2938e377c2e5) | ![CleanShot 2024-10-29 at 15 23 57](https://github.com/user-attachments/assets/d76086db-c272-4380-8a6a-59fc1b9abc34) |

Another thing i can see from your report is that your device was pretty low on memory when the ANR happens, only 200mb left. The code i added might introduced issues with high memory, but then, that code never gets executed in the bug's flow (it only gets executed when opening the search page):

 
![CleanShot 2024-10-29 at 15 28 40](https://github.com/user-attachments/assets/52fe8945-0adb-4f2e-b176-f923f373d43a)

**Comment by hannojg** [comment]
Aha, interesting, i can reproduce on release, but not on debug

**Comment by hannojg** [comment]
I can reproduce this with the adhoc build as well: 

- https://github.com/Expensify/App/pull/51590#issuecomment-2444662255

**Comment by hannojg** [comment]
I think the issue is caused by our native stack PR/bridgeless PR/or some general bug in reanimated that we've only discovered  now for some reason

**Comment by jasperhuangg** [comment]
@hannojg Hmm, shoot. I tried reproducing the bug for nearly 10 minutes on the AdHoc build and couldn't get it, but I was able to reproduce it within a minute on staging. I'm so sorry if I made a mistake reverting your PR.

Feel free to revert the revert again 😅 and I'd be happy to help you merge it. If this bug is really hard to reproduce, it makes sense that we could only find it now.

Do you have any recommendations on which PRs specifically would be causing this regression?

**Comment by jasperhuangg** [comment]
Ah hmm, that being said, @NikkiWines mentioned that the revert solved a deploy blocker that she was assigned to https://github.com/Expensify/App/issues/51527, so we may still want to investigate that. It also doesn't really seem like your code is invoked there, so it may be unrelated and just flaky like this one.

**Comment by hannojg** [comment]
Yes, i will make a new PR that addresses the other blockers soon!

For the crash: i can reproduce the bug very easily 100% of the time, so it should be easy for me to investigate. I can test a few likely PRs and revert them tmrw!

(I think you guys have a set of git commands that you use to revert a PR right, if so, could you share that one please?)

**Comment by Beamanator** [comment]
I'm marking this NAB since @hannojg confirmed this can be reproduced on older builds, cool? 👍 

Thanks so much for your investigation, both of y'all! 🙏

**Comment by hannojg** [comment]
Let us know if you want us to investigate whats going on (feel free to assign to me, either i will work on it or someone else from our margelo team can take a look!)
